test/jdk/sun/security/ssl/SSLEngineImpl/TestBadDNForPeerCA12.java

```diff
@@ -0,0 +1,203 @@
+        + "iL4yhg5vEZxXjny/G5Pz7DBXwWSm92WXwggWZ409G/7UT6VdsRLoJI0JVpn+UWE+B2Bg"
+        + "qV43ux+ymLPnQMbh2pp+4QBpfnForU1ZcnFNHPRE4f8hTOLK4NbqOmGuWCK3j7c0LOnq"
+        + "8/2EAlDxWFyPSfXkEochAwAAHADAB0gUsYb5i8nNJxMG2c6TY4P6TgXukhGhn2EqtOMC"
+        + "MYP1hUEAwBIMEYCIQCZN6uqbBUVnbWEVGacgAi8Inz+vbYB+nXxzXpS7B/rHgIhALedP"
+        + "TzefR4lNrQ5d4V42z0VIgIAEh811usG1hg58iFmDQAA7wNAAQIAKgQDBQMGAwgHCAgIB"
+        + "AgFCAYICQgKCAsEAQUBBgEEAgMDAwEDAgIDAgECAgC9AD0wOzELMAkRA1UEBhMCVVMxD"
+        + "TALBgNVBAoMBEphdmExHTAbBgNVBAsMFFN1bkpTU0UgVGVzdCBTZXJpdmNlAD0wOzELM"
+        + "AkRA1UEBhMCVVMxDTALBgNVBAoMBEphdmExHTAbBgNVBAsMFFN1bkpTU0UgVGVzdCBTZ"
+        + "XJpdmNlAD0wOzELMAkRA1UEBhMCVVMxDTALBgNVBAoMBEphdmExHTAbBgNVBAsMFFN1b"
+        + "kpTU0UgVGVzdCBTZXJpdmNlDgAAAA==");
+
+    /*
+     * The following is to set up the keystores.
+     */
+    private static final String passwd = "passphrase";
+
+    /*
+     * Main entry point for this demo.
+     */
+    public static void main(String[] args) throws Exception {
+        if (debug) {
+            System.setProperty("javax.net.debug", "all");
+        }
+
+        TestBadDNForPeerCA12 test = new TestBadDNForPeerCA12();
+
+        try {
+            test.runTest();
+            throw new Exception(
+                "TEST FAILED:  Didn't generate any exception");
+        } catch (SSLHandshakeException she) {
+            System.out.println("TEST PASSED:  Caught expected exception");
+        }
+    }
+
+    /*
+     * Create an initialized SSLContext to use for this demo.
+     */
+
+    public TestBadDNForPeerCA12() throws Exception {
+
+        KeyStore ks = KeyStore.getInstance("JKS");
+        KeyStore ts = KeyStore.getInstance("JKS");
+
+        char[] passphrase = passwd.toCharArray();
+
+        ks.load(new FileInputStream(keyStoreFile), passphrase);
+        ts.load(new FileInputStream(keyStoreFile), passphrase);
+
+        KeyManagerFactory kmf = KeyManagerFactory.getInstance("SunX509");
+        kmf.init(ks, passphrase);
+
+        TrustManagerFactory tmf = TrustManagerFactory.getInstance("SunX509");
+        tmf.init(ts);
+
+        SSLContext sslCtx = SSLContext.getInstance(proto);
+
+        sslCtx.init(kmf.getKeyManagers(), tmf.getTrustManagers(), null);
+
+        sslc = sslCtx;
+    }
+
+
+    private void runTest() throws Exception {
+
+        createSSLEngines();
+        createBuffers();
+
+        System.out.println("forcing client hello");
+        //sTOc = ByteBuffer.wrap(serverHello);
+        SSLEngineResult clientResult = clientEngine.wrap(clientOut, cTOs);
+        runDelegatedTasks(clientResult, clientEngine);
+
+        cTOs.flip();
+
+        sTOc = ByteBuffer.wrap(serverPayload);
+
+        SSLEngineResult clientHelloResult = clientEngine.unwrap(sTOc, clientIn);
+        System.out.println("client unwrap: " + clientHelloResult);
+        runDelegatedTasks(clientHelloResult, clientEngine);
+
+        sTOc.compact();
+        cTOs.compact();
```

Comment:
It doesn't really matter since the code will bomb out, but I don't think this line is doing what you expected.  cTOs is already pointing at the beginning of the Buffer.